### PR TITLE
IOS-9445 Hide last divider

### PR DESCRIPTION
**Pull request by pbartolome**
## 🎟️ **Jira ticket**
IOS-9445

## 🥅 **What's the goal?**
Hide last divider in mistica lists

## 🚧 **How do we do it?**
- Add a modifier to hide the cell divider
- Update usages
- Update documentation

## 🧪 **How can I verify this?**
Before:

https://github.com/Telefonica/mistica-ios/assets/2429905/62fcd302-31c0-4a39-8ddd-05522587c822

After:

https://github.com/Telefonica/mistica-ios/assets/2429905/99220923-3fa1-4aeb-8583-65c3a2e9b226

### CODE REVIEW

File: Sources/MisticaSwiftUI/Components/List/Cell.swift [hunk 1]
```diff
@@ -195,27 +198,6 @@ public struct Cell<PresetView: View, HeadlineView: View, Destination: View>: Vie
...
-            description: description,
-            assetType: assetType,
-            presetView: { presetView },
-            headlineView: { headlineView },
-            destinationView: destination
-        )
-        cell.allowsPressing = true
-        return cell
-    }
-}
```
Comment (pbartolome) [inline Sources/MisticaSwiftUI/Components/List/Cell.swift:217]: this modifier was breaking the chaining of modifiers... as it's creating a new `Cell`, it was removing any customisation made to the cell were the modifier was being applied (like for example, hiding the divider and then using this `navigationLink` will create again a new cell with divider)

File: Sources/MisticaSwiftUI/Components/List/Cell.swift [hunk 2]
```diff
@@ -418,27 +406,9 @@ struct ListSeparatorHiddenModifieriOS15: ViewModifier {
...
-
 public extension View {
     func listSeparatorHidden() -> some View {
-        conditionalModifier(
-            .iOS15,
-            then: { $0.modifier(ListSeparatorHiddenModifieriOS15()) },
-            else: { $0.modifier(ListSeparatorHiddenModifier()) }
-        )
```
Comment (pbartolome) [inline Sources/MisticaSwiftUI/Components/List/Cell.swift:441]: this modifier was not needed as the existing `ListSeparatorHiddenModifieriOS15` was already doing the `#available(iOS 15.0, *)` check

**Review by DavidMarinCalleja — APPROVED** [review]
good job!

**Review by dhidalgofadrique — APPROVED** [review]
👍

**Pull request merged**